### PR TITLE
feat: overload dump_write_request of replication_app_base.h

### PR DESCRIPTION
**Pull request by Shuo-Jia**
### What problem does this PR solve? 
https://github.com/XiaoMi/rdsn/pull/414 forbids the large write size request, but the log information is simple, https://github.com/XiaoMi/rdsn/pull/458 provide the interface of `dump_write_request`, this pr overload it.

### Check List 

Tests 
- Manual test (add detailed scripts or steps below)
  -  start onebox with config `max_allowed_write_size = 5`
  -  use shell `set` any key-value whose size > 5
  -  check the log:
```
[2.4@127.0.0.1:34801] client from 127.0.0.1:1 write request body size exceed threshold, request=[put:hash_key=a,sort_key=abc], request_body_size = 99, max_allowed_write_size = 5, it will be rejected!
``` 

Related changes

- Need to cherry-pick to the release branch(**depends on https://github.com/XiaoMi/rdsn/pull/458 merged**)
- Need to be included in the release note

### CODE REVIEW

File: src/server/pegasus_server_impl.cpp [hunk 1]
```diff
@@ -2618,5 +2618,63 @@ void pegasus_server_impl::release_db()
...
+
+    dsn::task_code rpc_code(request->rpc_code());
+    if (rpc_code == dsn::apps::RPC_RRDB_RRDB_PUT) {
+        auto put = put_rpc::auto_reply(request).request();
+        ::dsn::blob hash_key, sort_key;
+        pegasus_restore_key(put.key, hash_key, sort_key);
+        dwarn_replica("abnormal put: client_address = {}, hash_key = {}, sort_key = {}, put_bytes "
+                      "= {}, max_allowed_write_size = {}",
+                      request->header->from_address.to_std_string(),
+                      ::pegasus::utils::c_escape_string(hash_key).c_str(),
```
Comment (acelyc111) [inline src/server/pegasus_server_impl.cpp:2633]: dwarn_replica  is able to deal with std::string, so all `.c_str()` can be removed.

File: src/server/pegasus_server_impl.cpp [hunk 2]
```diff
@@ -2618,5 +2618,63 @@ void pegasus_server_impl::release_db()
...
+void pegasus_server_impl::dwarn_write_request(dsn::message_ex *request,
+                                              uint64_t max_allowed_write_size)
+{
+
+    dsn::task_code rpc_code(request->rpc_code());
+    if (rpc_code == dsn::apps::RPC_RRDB_RRDB_PUT) {
+        auto put = put_rpc::auto_reply(request).request();
+        ::dsn::blob hash_key, sort_key;
+        pegasus_restore_key(put.key, hash_key, sort_key);
+        dwarn_replica("abnormal put: client_address = {}, hash_key = {}, sort_key = {}, put_bytes "
```
Comment (levy5307) [inline src/server/pegasus_server_impl.cpp:2630]: `client from {} write request body size exceed threshold for this put operation`

File: src/server/pegasus_server_impl.cpp [hunk 3]
```diff
@@ -2618,5 +2618,57 @@ void pegasus_server_impl::release_db()
     _db = nullptr;
 }
 
+std::string pegasus_server_impl::dump_write_request(dsn::message_ex *request)
```
Comment (neverchanje) [inline src/server/pegasus_server_impl.cpp:2621]:
```suggestion
static std::string dump_write_request(dsn::message_ex *request)
```

I see no private member of pegasus_server_impl is used, so make it a static function.

File: src/server/pegasus_server_impl.cpp [hunk 4]
```diff
@@ -2618,5 +2618,57 @@ void pegasus_server_impl::release_db()
...
+    if (rpc_code == dsn::apps::RPC_RRDB_RRDB_PUT) {
+        auto put = put_rpc::auto_reply(request).request();
+        ::dsn::blob hash_key, sort_key;
+        pegasus_restore_key(put.key, hash_key, sort_key);
+        std::string request("put:");
+        request.append("hash_key=")
+            .append(pegasus::utils::c_escape_string(hash_key))
+            .append(",sort_key=")
+            .append(pegasus::utils::c_escape_string(sort_key));
+        return request;
```
Comment (neverchanje) [inline src/server/pegasus_server_impl.cpp:2633]:
```suggestion
        return fmt::format("hash_key={},sort_key={}", 
                            pegasus::utils::c_escape_string(hash_key), 
                            pegasus::utils::c_escape_string(sort_key));
```

File: src/server/pegasus_server_impl.cpp [hunk 5]
```diff
@@ -2618,5 +2618,57 @@ void pegasus_server_impl::release_db()
...
+    }
+
+    if (rpc_code == dsn::apps::RPC_RRDB_RRDB_MULTI_PUT) {
+        auto multi_put = multi_put_rpc::auto_reply(request).request();
+        std::string request("multi_put:");
+        request.append("hash_key=")
```
Comment (neverchanje) [inline src/server/pegasus_server_impl.cpp:2639]: using fmtlib will make it looks simpler and more efficient.

**Pull request merged**